File: src/types/Review.ts
/**
 * Review types based on the Android app and Firebase schema
 */

export interface DetailedRatings {
  quality: number;
  communication: number;
  timeliness: number;
  value: number;
  professionalism: number;
}

export interface OwnerResponse {
  content: string;
  respondedAt: number;
  ownerId: string;
}

export enum ReviewTargetType {
  SHOP = 'SHOP',
  PRODUCT = 'PRODUCT',
  SERVICE = 'SERVICE'
}

export enum ModerationStatus {
  PENDING = 'PENDING',
  APPROVED = 'APPROVED',
  REJECTED = 'REJECTED'
}

export interface Review {
  id: string;
  customerId: string;
  shopId: string;
  
  // Review Target
  targetType: ReviewTargetType;
  targetId: string;
  targetName: string;
  
  // Review Content
  rating: number;
  title: string;
  content: string;
  
  // Detailed Ratings
  detailedRatings: DetailedRatings;
  
  // Media
  imageUrls: string[];
  videoUrls: string[];
  
  // Context
  orderId: string;
  bookingId: string;
  verifiedPurchase: boolean;
  
  // Owner Response
  ownerResponse: OwnerResponse | null;
  
  // Helpfulness
  helpfulVotes: number;
  unhelpfulVotes: number;
  votedBy: string[];
  
  // Moderation
  visible: boolean;
  flagged: boolean;
  moderationStatus: ModerationStatus;
  moderationNotes: string;
  
  // Timestamps
  createdAt: number;
  updatedAt: number;
}

export interface CreateReviewRequest {
  shopId: string;
  targetType: ReviewTargetType;
  targetId: string;
  targetName: string;
  rating: number;
  title: string;
  content: string;
  detailedRatings?: Partial<DetailedRatings>;
  imageUrls?: string[];
  videoUrls?: string[];
  orderId?: string;
  bookingId?: string;
  verifiedPurchase?: boolean;
}

export interface ReviewStats {
  averageRating: number;
  totalReviews: number;
  ratingDistribution: {
    5: number;
    4: number;
    3: number;
    2: number;
    1: number;
  };
  detailedAverages: DetailedRatings;
}

File: androidAPP/MyHustle/app/src/main/java/com/blueclipse/myhustle/data/model/Review.kt
package com.blueclipse.myhustle.data.model

/**
 * Represents a review in the MyHustle platform
 */
data class Review(
    val id: String = "",
    val customerId: String = "",
    val shopId: String = "",
    
    // Review Target
    val targetType: ReviewTargetType = ReviewTargetType.SHOP,
    val targetId: String = "",
    val targetName: String = "",
    
    // Review Content
    val rating: Float = 0f,
    val title: String = "",
    val content: String = "",
    
    // Detailed Ratings
    val detailedRatings: DetailedRatings = DetailedRatings(),
    
    // Media
    val imageUrls: List<String> = emptyList(),
    val videoUrls: List<String> = emptyList(),
    
    // Context
    val orderId: String = "",
    val bookingId: String = "",
    val verifiedPurchase: Boolean = false,
    
    // Owner Response
    val ownerResponse: OwnerResponse? = null,
    
    // Helpfulness
    val helpfulVotes: Int = 0,
    val unhelpfulVotes: Int = 0,
    val votedBy: List<String> = emptyList(),
    
    // Moderation
    val visible: Boolean = true,
    val flagged: Boolean = false,
    val moderationStatus: ModerationStatus = ModerationStatus.APPROVED,
    val moderationNotes: String = "",
    
    // Timestamps
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    constructor() : this(
        "", "", "", ReviewTargetType.SHOP, "", "", 0f, "", "",
        DetailedRatings(), emptyList(), emptyList(), "", "", false,
        null, 0, 0, emptyList(), true, false, ModerationStatus.APPROVED, "",
        System.currentTimeMillis(), System.currentTimeMillis()
    )
}

/**
 * Detailed ratings for services
 */
data class DetailedRatings(
    val quality: Float = 0f,
    val communication: Float = 0f,
    val timeliness: Float = 0f,
    val value: Float = 0f,
    val professionalism: Float = 0f
) {
    constructor() : this(0f, 0f, 0f, 0f, 0f)
}

/**
 * Owner response to review
 */
data class OwnerResponse(
    val content: String = "",
    val respondedAt: Long = System.currentTimeMillis(),
    val ownerId: String = ""
) {
    constructor() : this("", System.currentTimeMillis(), "")
}

/**
 * Review target type enum
 */
enum class ReviewTargetType {
    SHOP,
    PRODUCT,
    SERVICE
}

/**
 * Moderation status enum
 */
enum class ModerationStatus {
    PENDING,
    APPROVED,
    REJECTED
}
